.github/workflows/openapi-update-generate-new-sdk.yml

```diff
@@ -35,11 +35,34 @@ jobs:
         run: |
           current_version=$(xmlstarlet sel -t -v "//*[local-name()='artifactVersion']" pom.xml)
           echo "Current version: $current_version"
-          IFS='.-' read -r -a parts <<< "$current_version"
-          new_patch=$((parts[2] + 1))
-          new_version="${parts[0]}.${parts[1]}.$new_patch-${parts[3]}"
+
+          unique_id=$(date +%Y%m%d%H%M%S)
+          if [[ "$current_version" == *"SNAPSHOT"* ]]; then
+            echo "Current version is a SNAPSHOT."
+            base_version=$(echo "$current_version" | sed -E 's/-SNAPSHOT-.*//')
+            new_version="${base_version}-SNAPSHOT-${unique_id}"
+          else
+            echo "Current version is the last release."
+            IFS='.-' read -r -a parts <<< "$current_version"
+            major=${parts[0]}
+            minor=${parts[1]}
+            patch=${parts[2]}
+            qualifier=${parts[3]}
+            
+            new_patch=$((patch + 1))
```

Comment:
I think this means all new versions will be patch versions. Are we OK with that?

Reply:
I thought about it, and I agree the increment needs human judgment. We can just add 'SNAPSHOT+commit number' automatically and then once we want to release the version will depends on the tag we are going to push manually, what do you think?
cc: @pierre-emmanuelJ

Reply:
The initial idea is to stay consistent with other java project like aws java sdk or like our internal Clojure repo versioning.
- `v0.0.1`
- `v0.0.2-SNAPSHOT-g147ef` a commit on master after a stable release.
- `v0.0.2-SNAPSHOT-de76re` a commit on master after a stable release.
- `v0.0.2` we trigger manually a release with a decided tag
- `v0.0.3-SNAPSHOT-re78r6` a commit on master after a stable release.
- `v0.1.0` we trigger manually a release with a decided tag
- `v0.1.1-SNAPSHOT-hjk433` a commit on master after a stable release.

Every commit will be a minor change, because we assume something changes and we add SNAPSHOT, and it's when we manually trigger release we can decide if there is a minor or breaking...etc.

Developer relying on snapshot know it's not a stable release...


The final goal will be able to point GitHub pages on Javadoc with versioned link to all stable release and only a link to the latest snapshot, we will not keep history on of all the snapshot in the doc ;)